src/main/java/de/doubleslash/keeptime/common/DateFormatter.java

```diff
@@ -55,4 +63,36 @@ public static String toTimeString(final LocalDateTime localDateTime) {
       return localDateTime.format(timeFormatter);
    }
 
+   public static void setSystemLocale(Locale locale) {
+      systemLocale = locale;
+   }
+
+   public static Locale getSystemLocale() {
+      return systemLocale;
+   }
+
+   public static void applySystemLocaleOnDate(DatePicker datePicker) {
+      DateTimeFormatter formatter = DateTimeFormatter.ofLocalizedDate(FormatStyle.SHORT).withLocale(systemLocale);
+
+      datePicker.setConverter(new StringConverter<>() {
+         @Override
+         public String toString(LocalDate date) {
+            return (date != null) ? formatter.format(date) : "";
+         }
+
+         @Override
+         public LocalDate fromString(String s) {
+            try {
+               return (s != null && !s.isEmpty()) ? LocalDate.parse(s, formatter) : null;
+            } catch (DateTimeParseException e) {
+               return null;
+            }
+         }
+      });
+
+      datePicker.setOnShowing(e-> Locale.setDefault(Locale.Category.FORMAT, systemLocale));
+      datePicker.setOnHiding(e-> Locale.setDefault(Locale.Category.FORMAT, Locale.getDefault()));
+      datePicker.setOnAction(e-> Locale.setDefault(Locale.Category.FORMAT, Locale.getDefault()));
```

Comment:
these resets did not make sense for me, as you reset it to the "wrong" locale. But it showed me this way to only set the locale for FORMAT. I created #191 to most likely replace this PR - lets review it.